tests/gcp_pilot/identity_platform_test.py

```diff
@@ -1,8 +1,38 @@
 import unittest
 
-from gcp_pilot.identity_platform import IdentityPlatform
+from gcp_pilot.factories.identity_platform import FirebaseAuthTokenFactory
+from gcp_pilot.identity_platform import IdentityPlatform, FirebaseAuthToken, parse_timestamp
+from gcp_pilot.mocker import patch_firebase_token
 from tests import ClientTestMixin
 
 
 class TestIdentityPlatform(ClientTestMixin, unittest.TestCase):
     _CLIENT_KLASS = IdentityPlatform
+
+    def test_parse_firebase_token(self):
+        token_data = FirebaseAuthTokenFactory.create()
+        with patch_firebase_token(return_value=token_data):
+            token = FirebaseAuthToken(jwt_token="potato")
+
+        self.assertEqual(parse_timestamp(token_data["exp"]), token.expiration_date)
+        self.assertEqual(token_data["event_type"], token.event_type)
+        self.assertEqual(token_data["ip_address"], token.ip_address)
+        self.assertEqual(token_data["sign_in_method"], token.provider_id)
+        # self.assertEqual(token_data["tenant_id"], token.tenant_id)  # FIXME
```

Comment:
What's wrong with this one?

Reply:
I still couldn't reproduce this use case in the real world to be sure of the field name.